update fisher_swbd recipe

naxingyu opened the pull request:
update fisher_swbd to adopt new path and srilm installer.

jtrmal commented on tools/extras/install_srilm.sh:
```diff
@@ -14,7 +14,7 @@ if [ ! -d liblbfgs-1.10 ]; then
 
 # http://www.speech.sri.com/projects/srilm/download.html
-if [ ! -f srilm.tgz ]; then
+if [ ! -f srilm-1.7.2.tar ]; then
```
@naxingyu , I won't comment on the recipe changes per se (not familiar with that)
But I'm not fond of the changes here.
a) what srilm pages do offer for download if you go there is srilm-1.7.2.tar.gz, i.e. I'm not sure where the tar file comes from, but that script won't work for other people now.
b) I'm still in favor of asking users to do a symlink or copy the file as srilm.tgz.
I think it makes sense as we cannot download the file automatically -- so people can use whatever version they have available and do not have to undergo the downloading process every time.  Which I realize it's not true now as I'm thinking about it, as we need 1.7.1 at least, because of the maxent LMs. So I don't really object to having a fixed version and bump it now and then, but I think it's less convenient for us.
c) "$SRILM" is preferred, IMO.
 It's because of cases where the users have space (white character) in the path somewhere. Or perhaps not, but is there any rationale behind changing it?

naxingyu commented:
> 在 2017年3月17日，23:58，jtrmal <notifications@github.com> 写道：
> 
> @jtrmal commented on this pull request.
> 
> In tools/extras/install_srilm.sh <https://github.com/kaldi-asr/kaldi/pull/1498#discussion_r106684126>:
> 
> > @@ -14,7 +14,7 @@ if [ ! -d liblbfgs-1.10 ]; then
>  fi
>  
>  # http://www.speech.sri.com/projects/srilm/download.html
> -if [ ! -f srilm.tgz ]; then
> +if [ ! -f srilm-1.7.2.tar ]; then
> @naxingyu <https://github.com/naxingyu> , I won't comment on the recipe changes per se (not familiar with that)
> But I'm not fond of the changes here.
> a) what srilm pages do offer for download if you go there is srilm-1.7.2.tar.gz, i.e. I'm not sure where the tar file comes from, but that script won't work for other people now.
> 
The tar file is what I got from the download page, last month. I don’t if they have changed the offered file from tgz to tar, but that’s what I got.
> b) I'm still in favor of asking users to do a symlink or copy the file as srilm.tgz.
> I think it makes sense as we cannot download the file automatically -- so people can use whatever version they have available and do not have to undergo the downloading process every time. Which I realize it's not true now as I'm thinking about it, as we need 1.7.1 at least, because of the maxent LMs. So I don't really object to having a fixed version and bump it now and then, but I think it's less convenient for us.
> c) "$SRILM" is preferred, IMO.
> 
I agree. Will change it back.
> It's because of cases where the users have space (white character) in the path somewhere. Or perhaps not, but is there any rationale behind changing it?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/kaldi-asr/kaldi/pull/1498#pullrequestreview-27621733>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ADKpxJ5ZfhBIIqv5VCw8TsRyBIDBVhr3ks5rmq2rgaJpZM4MgSPH>.
>

danpovey commented:
Check with a couple ways of getting it, maybe some browsers or OS's
automatically unzip things.


On Fri, Mar 17, 2017 at 9:07 PM, Xingyu Na <notifications@github.com> wrote:

>
> > 在 2017年3月17日，23:58，jtrmal <notifications@github.com> 写道：
> >
> > @jtrmal commented on this pull request.
> >
> > In tools/extras/install_srilm.sh <https://github.com/kaldi-asr/
> kaldi/pull/1498#discussion_r106684126>:
> >
> > > @@ -14,7 +14,7 @@ if [ ! -d liblbfgs-1.10 ]; then
> > fi
> >
> > # http://www.speech.sri.com/projects/srilm/download.html
> > -if [ ! -f srilm.tgz ]; then
> > +if [ ! -f srilm-1.7.2.tar ]; then
> > @naxingyu <https://github.com/naxingyu> , I won't comment on the recipe
> changes per se (not familiar with that)
> > But I'm not fond of the changes here.
> > a) what srilm pages do offer for download if you go there is
> srilm-1.7.2.tar.gz, i.e. I'm not sure where the tar file comes from, but
> that script won't work for other people now.
> >
> The tar file is what I got from the download page, last month. I don’t if
> they have changed the offered file from tgz to tar, but that’s what I got.
> > b) I'm still in favor of asking users to do a symlink or copy the file
> as srilm.tgz.
> > I think it makes sense as we cannot download the file automatically --
> so people can use whatever version they have available and do not have to
> undergo the downloading process every time. Which I realize it's not true
> now as I'm thinking about it, as we need 1.7.1 at least, because of the
> maxent LMs. So I don't really object to having a fixed version and bump it
> now and then, but I think it's less convenient for us.
> > c) "$SRILM" is preferred, IMO.
> >
> I agree. Will change it back.
> > It's because of cases where the users have space (white character) in
> the path somewhere. Or perhaps not, but is there any rationale behind
> changing it?
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/1498#pullrequestreview-27621733>,
> or mute the thread <https://github.com/notifications/unsubscribe-auth/
> ADKpxJ5ZfhBIIqv5VCw8TsRyBIDBVhr3ks5rmq2rgaJpZM4MgSPH>.
>
> >
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/1498#issuecomment-287505477>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVuxgsXPeop0EocRZWQsWiZpkXL3w1ks5rmy5ugaJpZM4MgSPH>
> .
>

naxingyu commented:
You are right… I’ll change the extension. And do I keep the version numbers or not?

> 在 2017年3月18日，09:12，Daniel Povey <notifications@github.com> 写道：
> 
> Check with a couple ways of getting it, maybe some browsers or OS's
> automatically unzip things.
> 
> 
> On Fri, Mar 17, 2017 at 9:07 PM, Xingyu Na <notifications@github.com> wrote:
> 
> >
> > > 在 2017年3月17日，23:58，jtrmal <notifications@github.com> 写道：
> > >
> > > @jtrmal commented on this pull request.
> > >
> > > In tools/extras/install_srilm.sh <https://github.com/kaldi-asr/
> > kaldi/pull/1498#discussion_r106684126>:
> > >
> > > > @@ -14,7 +14,7 @@ if [ ! -d liblbfgs-1.10 ]; then
> > > fi
> > >
> > > # http://www.speech.sri.com/projects/srilm/download.html
> > > -if [ ! -f srilm.tgz ]; then
> > > +if [ ! -f srilm-1.7.2.tar ]; then
> > > @naxingyu <https://github.com/naxingyu> , I won't comment on the recipe
> > changes per se (not familiar with that)
> > > But I'm not fond of the changes here.
> > > a) what srilm pages do offer for download if you go there is
> > srilm-1.7.2.tar.gz, i.e. I'm not sure where the tar file comes from, but
> > that script won't work for other people now.
> > >
> > The tar file is what I got from the download page, last month. I don’t if
> > they have changed the offered file from tgz to tar, but that’s what I got.
> > > b) I'm still in favor of asking users to do a symlink or copy the file
> > as srilm.tgz.
> > > I think it makes sense as we cannot download the file automatically --
> > so people can use whatever version they have available and do not have to
> > undergo the downloading process every time. Which I realize it's not true
> > now as I'm thinking about it, as we need 1.7.1 at least, because of the
> > maxent LMs. So I don't really object to having a fixed version and bump it
> > now and then, but I think it's less convenient for us.
> > > c) "$SRILM" is preferred, IMO.
> > >
> > I agree. Will change it back.
> > > It's because of cases where the users have space (white character) in
> > the path somewhere. Or perhaps not, but is there any rationale behind
> > changing it?
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub <
> > https://github.com/kaldi-asr/kaldi/pull/1498#pullrequestreview-27621733>,
> > or mute the thread <https://github.com/notifications/unsubscribe-auth/
> > ADKpxJ5ZfhBIIqv5VCw8TsRyBIDBVhr3ks5rmq2rgaJpZM4MgSPH>.
> >
> > >
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/kaldi-asr/kaldi/pull/1498#issuecomment-287505477>, or mute
> > the thread
> > <https://github.com/notifications/unsubscribe-auth/ADJVuxgsXPeop0EocRZWQsWiZpkXL3w1ks5rmy5ugaJpZM4MgSPH>
> > .
> >
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/kaldi-asr/kaldi/pull/1498#issuecomment-287505799>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ADKpxErNlWoxchB600GY5HlGzSh409aBks5rmy9tgaJpZM4MgSPH>.
>

danpovey commented:
Let's not keep version numbers, it will be more compatible with existing
setups.

On Fri, Mar 17, 2017 at 9:15 PM, Xingyu Na <notifications@github.com> wrote:

> You are right… I’ll change the extension. And do I keep the version
> numbers or not?
>
>
> > 在 2017年3月18日，09:12，Daniel Povey <notifications@github.com> 写道：
> >
> > Check with a couple ways of getting it, maybe some browsers or OS's
> > automatically unzip things.
> >
> >
> > On Fri, Mar 17, 2017 at 9:07 PM, Xingyu Na <notifications@github.com>
> wrote:
> >
> > >
> > > > 在 2017年3月17日，23:58，jtrmal <notifications@github.com> 写道：
> > > >
> > > > @jtrmal commented on this pull request.
> > > >
> > > > In tools/extras/install_srilm.sh <https://github.com/kaldi-asr/
> > > kaldi/pull/1498#discussion_r106684126>:
> > > >
> > > > > @@ -14,7 +14,7 @@ if [ ! -d liblbfgs-1.10 ]; then
> > > > fi
> > > >
> > > > # http://www.speech.sri.com/projects/srilm/download.html
> > > > -if [ ! -f srilm.tgz ]; then
> > > > +if [ ! -f srilm-1.7.2.tar ]; then
> > > > @naxingyu <https://github.com/naxingyu> , I won't comment on the
> recipe
> > > changes per se (not familiar with that)
> > > > But I'm not fond of the changes here.
> > > > a) what srilm pages do offer for download if you go there is
> > > srilm-1.7.2.tar.gz, i.e. I'm not sure where the tar file comes from,
> but
> > > that script won't work for other people now.
> > > >
> > > The tar file is what I got from the download page, last month. I don’t
> if
> > > they have changed the offered file from tgz to tar, but that’s what I
> got.
> > > > b) I'm still in favor of asking users to do a symlink or copy the
> file
> > > as srilm.tgz.
> > > > I think it makes sense as we cannot download the file automatically
> --
> > > so people can use whatever version they have available and do not have
> to
> > > undergo the downloading process every time. Which I realize it's not
> true
> > > now as I'm thinking about it, as we need 1.7.1 at least, because of the
> > > maxent LMs. So I don't really object to having a fixed version and
> bump it
> > > now and then, but I think it's less convenient for us.
> > > > c) "$SRILM" is preferred, IMO.
> > > >
> > > I agree. Will change it back.
> > > > It's because of cases where the users have space (white character) in
> > > the path somewhere. Or perhaps not, but is there any rationale behind
> > > changing it?
> > > >
> > > > —
> > > > You are receiving this because you were mentioned.
> > > > Reply to this email directly, view it on GitHub <
> > > https://github.com/kaldi-asr/kaldi/pull/1498#
> pullrequestreview-27621733>,
> > > or mute the thread <https://github.com/notifications/unsubscribe-auth/
> > > ADKpxJ5ZfhBIIqv5VCw8TsRyBIDBVhr3ks5rmq2rgaJpZM4MgSPH>.
> > >
> > > >
> > >
> > > —
> > > You are receiving this because you are subscribed to this thread.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/kaldi-asr/kaldi/pull/1498#issuecomment-287505477>,
> or mute
> > > the thread
> > > <https://github.com/notifications/unsubscribe-auth/
> ADJVuxgsXPeop0EocRZWQsWiZpkXL3w1ks5rmy5ugaJpZM4MgSPH>
> > > .
> > >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/1498#issuecomment-287505799>, or
> mute the thread <https://github.com/notifications/unsubscribe-auth/
> ADKpxErNlWoxchB600GY5HlGzSh409aBks5rmy9tgaJpZM4MgSPH>.
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/1498#issuecomment-287506084>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu-w6l4i9JnaCpAR1l1hq756OIUziks5rmzBOgaJpZM4MgSPH>
> .
>

naxingyu commented:
OK. I’ve reverted the changes on installation script.

> 在 2017年3月18日，09:18，Daniel Povey <notifications@github.com> 写道：
> 
> Let's not keep version numbers, it will be more compatible with existing
> setups.
> 
> On Fri, Mar 17, 2017 at 9:15 PM, Xingyu Na <notifications@github.com> wrote:
> 
> > You are right… I’ll change the extension. And do I keep the version
> > numbers or not?
> >
> >
> > > 在 2017年3月18日，09:12，Daniel Povey <notifications@github.com> 写道：
> > >
> > > Check with a couple ways of getting it, maybe some browsers or OS's
> > > automatically unzip things.
> > >
> > >
> > > On Fri, Mar 17, 2017 at 9:07 PM, Xingyu Na <notifications@github.com>
> > wrote:
> > >
> > > >
> > > > > 在 2017年3月17日，23:58，jtrmal <notifications@github.com> 写道：
> > > > >
> > > > > @jtrmal commented on this pull request.
> > > > >
> > > > > In tools/extras/install_srilm.sh <https://github.com/kaldi-asr/
> > > > kaldi/pull/1498#discussion_r106684126>:
> > > > >
> > > > > > @@ -14,7 +14,7 @@ if [ ! -d liblbfgs-1.10 ]; then
> > > > > fi
> > > > >
> > > > > # http://www.speech.sri.com/projects/srilm/download.html
> > > > > -if [ ! -f srilm.tgz ]; then
> > > > > +if [ ! -f srilm-1.7.2.tar ]; then
> > > > > @naxingyu <https://github.com/naxingyu> , I won't comment on the
> > recipe
> > > > changes per se (not familiar with that)
> > > > > But I'm not fond of the changes here.
> > > > > a) what srilm pages do offer for download if you go there is
> > > > srilm-1.7.2.tar.gz, i.e. I'm not sure where the tar file comes from,
> > but
> > > > that script won't work for other people now.
> > > > >
> > > > The tar file is what I got from the download page, last month. I don’t
> > if
> > > > they have changed the offered file from tgz to tar, but that’s what I
> > got.
> > > > > b) I'm still in favor of asking users to do a symlink or copy the
> > file
> > > > as srilm.tgz.
> > > > > I think it makes sense as we cannot download the file automatically
> > --
> > > > so people can use whatever version they have available and do not have
> > to
> > > > undergo the downloading process every time. Which I realize it's not
> > true
> > > > now as I'm thinking about it, as we need 1.7.1 at least, because of the
> > > > maxent LMs. So I don't really object to having a fixed version and
> > bump it
> > > > now and then, but I think it's less convenient for us.
> > > > > c) "$SRILM" is preferred, IMO.
> > > > >
> > > > I agree. Will change it back.
> > > > > It's because of cases where the users have space (white character) in
> > > > the path somewhere. Or perhaps not, but is there any rationale behind
> > > > changing it?
> > > > >
> > > > > —
> > > > > You are receiving this because you were mentioned.
> > > > > Reply to this email directly, view it on GitHub <
> > > > https://github.com/kaldi-asr/kaldi/pull/1498#
> > pullrequestreview-27621733>,
> > > > or mute the thread <https://github.com/notifications/unsubscribe-auth/
> > > > ADKpxJ5ZfhBIIqv5VCw8TsRyBIDBVhr3ks5rmq2rgaJpZM4MgSPH>.
> > > >
> > > > >
> > > >
> > > > —
> > > > You are receiving this because you are subscribed to this thread.
> > > > Reply to this email directly, view it on GitHub
> > > > <https://github.com/kaldi-asr/kaldi/pull/1498#issuecomment-287505477>,
> > or mute
> > > > the thread
> > > > <https://github.com/notifications/unsubscribe-auth/
> > ADJVuxgsXPeop0EocRZWQsWiZpkXL3w1ks5rmy5ugaJpZM4MgSPH>
> > > > .
> > > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub <
> > https://github.com/kaldi-asr/kaldi/pull/1498#issuecomment-287505799>, or
> > mute the thread <https://github.com/notifications/unsubscribe-auth/
> > ADKpxErNlWoxchB600GY5HlGzSh409aBks5rmy9tgaJpZM4MgSPH>.
> > >
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/kaldi-asr/kaldi/pull/1498#issuecomment-287506084>, or mute
> > the thread
> > <https://github.com/notifications/unsubscribe-auth/ADJVu-w6l4i9JnaCpAR1l1hq756OIUziks5rmzBOgaJpZM4MgSPH>
> > .
> >
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/kaldi-asr/kaldi/pull/1498#issuecomment-287506282>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ADKpxNopo4-FtebdIzr0xePIYd3zJgtYks5rmzDpgaJpZM4MgSPH>.
>

danpovey commented on egs/fisher_swbd/s5/local/fisher_train_lms.sh:
```diff
@@ -20,32 +20,16 @@ done
 
+kaldi_lm=`which train_lm.sh`
+if [ ! -x $kaldi_lm ]; then
+  echo "train_lm.sh is not found. Checkout tools/extra/install_kaldi_lm.sh"
```
Say "look at" not "checkout".

naxingyu commented:
Changed and squashed.

> 在 2017年3月18日，09:27，Daniel Povey <notifications@github.com> 写道：
> 
> @danpovey commented on this pull request.
> 
> In egs/fisher_swbd/s5/local/fisher_train_lms.sh <https://github.com/kaldi-asr/kaldi/pull/1498#discussion_r106769168>:
> 
> >  
> +kaldi_lm=`which train_lm.sh`
> +if [ ! -x $kaldi_lm ]; then
> +  echo "train_lm.sh is not found. Checkout tools/extra/install_kaldi_lm.sh"
> Say "look at" not "checkout".
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/kaldi-asr/kaldi/pull/1498#pullrequestreview-27713317>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ADKpxHIJMklde-LQR7tukrbog1_cl0f2ks5rmzL_gaJpZM4MgSPH>.
>

danpovey commented:
Merging. Thanks!
